### ISSUE TITLE
Cancel internal miner solver tasks when the best chain tip changes

### ISSUE THREAD

**Issue by arya2** [ZcashFoundation/zebra#8797]
## Motivation

When mining with many solver tasks, Zebra's miner can create many chain forks on Testnet (10 were seen on the NU6 private Testnet).

##### Related

See if there are any changes in https://github.com/ZcashFoundation/zebra/pull/8174 that may still be valuable.

**Comment by arya2** [comment]
This isn't too important, forks in test environments are okay, and it's currently set to only use 1 thread so addressing this issue would enable running multiple mining threads in parallel, which is rarely useful. We should do this cleanup after zcashd deprecation.

**Comment by mpguerra** [comment]
not a priority right now
